sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -57,6 +62,39 @@
         this.serviceVersion = serviceVersion;
     }
 
+    private static Mono<PollResponse<OperationResult>> processAnalyzeModelResponse(
+        SimpleResponse<AnalyzeOperationResult> analyzeOperationResultSimpleResponse,
+        PollResponse<OperationResult> operationResultPollResponse) {
+        LongRunningOperationStatus status;
+        switch (analyzeOperationResultSimpleResponse.getValue().getStatus()) {
+            case NOT_STARTED:
+            case RUNNING:
+                status = LongRunningOperationStatus.IN_PROGRESS;
+                break;
+            case SUCCEEDED:
+                status = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
+                break;
+            case FAILED:
+                status = LongRunningOperationStatus.FAILED;
+                break;
+            default:
+                status = LongRunningOperationStatus.fromString(
+                    analyzeOperationResultSimpleResponse.getValue().getStatus().toString(), true);
```

Comment:
Why is the default case to assume the operation is complete? Is this a case we should never hit?

Reply:
Yes, the above-defined cases should cover all of the `OperationStatus` statuses and this should never be hit. But if it does it should complete the LRO with the status it found.